src/shimmerHttp.js

```diff
@@ -0,0 +1,215 @@
+import http from 'http';
+import https from 'https';
+import url from 'url';
+import { debuglog } from 'util';
+import shimmer from 'shimmer';
+import Perf from 'performance-node';
+import uuid from 'uuid/v4';
+import pickBy from 'lodash.pickby';
+import isArray from 'isarray';
+import { flatten } from 'flat';
+
+const debug = debuglog('@iopipe/trace');
+
+/*eslint-disable babel/no-invalid-this*/
+
+function unwrap() {
+  if (http.get.__wrapped) {
+    shimmer.unwrap(http, 'get');
+  }
+  if (http.request.__wrapped) {
+    shimmer.unwrap(http, 'request');
+  }
+  if (https.get.__wrapped) {
+    shimmer.unwrap(https, 'get');
+  }
+  if (https.request.__wrapped) {
+    shimmer.unwrap(https, 'request');
+  }
+  delete http.__iopipeShimmer;
+  delete https.__iopipeShimmer;
+}
+
+function wrapHttpGet(mod) {
+  return (options, cb) => {
+    const req = mod.request(options, cb);
+    req.end();
+    return req;
+  };
+}
+
+// these are keys that are node specific and come from the actual js request object
+const unnecessaryReqKeys = [
+  'agent',
+  'automaticFailover',
+  'cache',
+  'decompress',
+  'followRedirect',
+  'retries',
+  'slashes',
+  'strictTtl',
+  'throwHttpErrors',
+  'useElectronNet'
+];
+
+function excludeUnnecessaryReqKeys(obj) {
+  return pickBy(obj, (v, k) => unnecessaryReqKeys.indexOf(k) === -1);
+}
+
+function getReqDataObject(rawOptions) {
+  // options might be a string ie http.get(), coerce to object
+  const data =
+    typeof rawOptions === 'string'
+      ? { href: rawOptions, headers: {} }
+      : Object.assign({}, rawOptions) || {};
+
+  const urlObject =
+    typeof data === 'string' ? url.parse(rawOptions) : rawOptions;
+  Object.assign(data, urlObject);
+
+  // ensure url key is present with full URI
+  data.url = data.href || url.format(data);
+  // simple rename
+  data.query = data.search;
+  // delete duplicate or extraneous keys
+  delete data.search;
+  delete data.host;
+  delete data.href;
+  // delete data['user-agent'];
+
+  // sometimes request headers come in as an array
+  // make them strings to conform to our schema better
+  Object.keys(data.headers || {}).forEach(k => {
+    data.headers[k] = isArray(data.headers[k])
+      ? data.headers[k].join(' ')
+      : data.headers[k];
+  });
+
+  return excludeUnnecessaryReqKeys(data);
```

Comment:
If you have a map object like above, you can just ignore any keys that aren't in your map.

Reply:
@kolanos  Maybe need some clarification here

Reply:
I'm not 100% sure what this means but I am making a change that eliminates a few lines of code to be cleaner that I think is what you're looking for.